pkgs/by-name/lu/lux-cli/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  libgit2,
+  gnupg,
+  libgpg-error,
+  gpgme,
+  darwin,
+  installShellFiles,
+  luajit,
+  makeWrapper,
+  symlinkJoin,
+  lua51Packages,
+  lua52Packages,
+  lua53Packages,
+  lua54Packages,
+  nix,
+  versionCheckHook,
+}:
+let
+  lux-lua-bundle = symlinkJoin {
+    name = "lux-lua-bundle";
+    paths = [
+      lua51Packages.lux-lua
+      lua52Packages.lux-lua
+      lua53Packages.lux-lua
+      lua54Packages.lux-lua
+    ];
+  };
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lux-cli";
+
+  version = "0.3.13";
+
+  src = fetchFromGitHub {
+    owner = "nvim-neorocks";
+    repo = "lux";
+    rev = "v${version}";
+    hash = "sha256-G/j6Tz6mZwdpo7QmVF7Vw5BrEhTtKtzoLh0qkWXxjsQ=";
+  };
+
+  buildAndTestSubdir = "lux-cli";
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-duyaLqwiprrGATTA9ZmvzNaOhnrq1EdlCNobBUCFYUc=";
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgramArg = "--version";
+  doInstallCheck = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+    pkg-config
+    installShellFiles
+  ];
+
+  buildInputs =
+    [
+      luajit
+      openssl
+      libgit2
+      gnupg
+      libgpg-error
+      gpgme
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.SystemConfiguration
+    ];
+
+  env = {
+    LIBGIT2_NO_VENDOR = 1;
+    LIBSSH2_SYS_USE_PKG_CONFIG = 1;
+    LUX_SKIP_IMPURE_TESTS = 1; # Disable impure unit tests
+  };
+
+  cargoTestFlags = "--lib"; # Disable impure integration tests
+
+  nativeCheckInputs = [
+    luajit
+    nix
+  ];
+
+  postBuild = ''
+    cargo xtask dist-man
+    cargo xtask dist-completions
+  '';
+
+  postFixup = ''
+    # Instruct Lux to search for the lux-specific shared libraries in the lux-lua bundle
+    # (temporary solution, until https://github.com/nvim-neorocks/lux/issues/655 is implemented)
+    wrapProgram $out/bin/lx --set LUX_LIB_DIR "${lux-lua-bundle}"
+  '';
+
+  meta = with lib; {
```

Comment:
Please avoid `with lib`

Reply:
Is this documented somewhere ? It usd to be the standard.